app/Repositories/CourseSettingsRepository.php

```diff
@@ -46,34 +64,71 @@ public function getCourseSettings($courseId)
             'courseCategory',
             'courseCategory.category',
             'courseFilter',
-            'courseFilter.filter'
+            'courseFilter.filter',
+            'image'
         ])->where('course_id', $courseId)->first();
     }
 
-    public function updateCourseSettings(int $courseId, CourseSettingsRequest $courseSettings)
+    public function getCourseSettingsForAllCourses()
+    {
+        return CourseSettings::with([
+            'courseCategory',
+            'courseCategory.category',
+            'courseFilter',
+            'courseFilter.filter',
+            'image'
+        ])->get();
+    }
+
+    public function getCourseImages(){
+        return CourseImage::all();
+    }
+
+    public function addFilter(FilterRequest $filter)
+    {
+        $filter = Filter::create([
+            'filter_name' => $filter['filter_name'],
+            'type' => $filter['type'],
+        ]);
+        return $filter;
+    }
+
+    public function addCategory(CategoryRequest $category)
     {
+        $category = Category::create([
+            'name' => $category['name'],
+            'position' => $category['position'],
+            'color_code' => $category['color_code'],
+        ]);
+        return $category;
+    }
 
+    public function updateCourseSettings(int $courseId, CourseSettingsRequest $courseSettings)
+    {
         CourseSettings::updateOrCreate(
             ['course_id' => $courseId],
             [
                 'course_id' => $courseId,
-                'unmaintained_since' => new DateTime($courseSettings['unmaintained_since']),
+                'unmaintained_since' => $courseSettings['unmaintained_since']? new DateTime($courseSettings['unmaintained_since']) : null,
                 'role_support' => $courseSettings['role_support'],
                 'licence' => $courseSettings['licence'],
                 'multilang' => $courseSettings['multilang'],
                 'banner_type' => $courseSettings['banner_type'],
                 'banner_text' => $courseSettings['banner_text'],
+                'image_id' => $courseSettings['image_id']
             ]
         );
-        $newCourseCategory = $courseSettings['courseCategory'][0];
-        $courseCategory = CourseCategory::updateOrCreate(
-            ['course_id' => $courseId],
-            [
-                'category_id' => $newCourseCategory['category_id'],
-                'new' => $newCourseCategory['new'],
-                'position' => $newCourseCategory['position']
-            ]
-        );
+        if($courseSettings['courseCategory'] != null){
+            $newCourseCategory = $courseSettings['courseCategory'][0];
```

Comment:
public function updateCourseSettings(int $courseId, CourseSettingsRequest $courseSettings)
{
    // ...
    CourseSettings::updateOrCreate(
        // ...
    );

    if ($courseSettings['courseCategory'] != null) {
        $newCourseCategory = $courseSettings['courseCategory'][0];
        $courseCategory = CourseCategory::updateOrCreate(
            // ...
        );
    }

    $currentCourseFilters = CourseFilter::where('course_id', $courseId)->get();
    $courseFilters = $courseSettings['courseFilters'];
    // ...
}

In this code, you're creating instances of model classes CourseSettings, CourseCategory, and CourseFilter directly within the updateCourseSettings method. Instead of creating instances within the method, you can consider using dependency injection to pass these model instances to the repository's constructor, making your code more testable and following best practices.